docs: fix the is_deleted option docs (#49704)
ClickHouse/ClickHouse: docs/en/engines/table-engines/mergetree-family/replacingmergetree.md

### is_deleted

> `is_deleted` —  Name of a `UInt8` column with the type of row: `1` is a “deleted“ row, `0` is a “state“ row.

Reviewer: ```suggestion
`is_deleted` —  Name of a column used during a merge to determine whether the data in this row represents the state or is to be deleted; `1` is a “deleted“ row, `0` is a “state“ row.
```

>     Column data type — `UInt8`.

Reviewer: The four leading spaces caused this to be rendered as a code block, two will indent it with the rest of the text without the code block.

>     Can only be enabled when `ver` is used.

Reviewer: ```suggestion
  
  :::note
  `is_deleted` can only be enabled when `ver` is used.
```

>     The row is deleted when the `OPTIMIZE ... FINAL CLEANUP`, or `OPTIMIZE ... FINAL` is used or if the engine settings `clean_deleted_rows` has been set to `Always`.

Reviewer: ```suggestion
  The row is deleted when the `OPTIMIZE ... FINAL CLEANUP`, or `OPTIMIZE ... FINAL` is used or if the engine settings `clean_deleted_rows` has been set to `Always`.
  :::
  
```
